File: main_lst_processing.py
import rasterio
import numpy as np
import os

# ---------------------------
# 1. Paths
# ---------------------------
input_file = 'data/landsat_band10.tif'
output_file = 'results/LST_Map.tif'

# Create results folder if it doesn't exist
os.makedirs('results', exist_ok=True)

# ---------------------------
# 2. Read Landsat Band 10
# ---------------------------
with rasterio.open(input_file) as src:
    DN = src.read(1).astype(float)
    profile = src.profile

# ---------------------------
# 3. Radiance → Brightness Temperature → LST
# ---------------------------
# Example constants (same as MATLAB version)
ML = 3.3420E-04
AL = 0.1
K1 = 774.8853
K2 = 1321.0789

# Radiance
radiance = ML * DN + AL

# Brightness Temperature
BT = K2 / np.log((K1 / radiance) + 1)

# LST (Celsius)
LST = BT - 273.15

# ---------------------------
# 4. Save LST map
# ---------------------------
profile.update(dtype=rasterio.float32, count=1)

with rasterio.open(output_file, 'w', **profile) as dst:
    dst.write(LST.astype(rasterio.float32), 1)

print(f"LST map saved to {output_file}")

File: src/main_lst_processing.m

% =========================================================
% Main Script: Land Surface Temperature (LST) Mapping
% =========================================================
clc; clear;

% Input thermal band (example name)
thermalBand = 'data/landsat_band10.tif';

% Radiometric calibration constants (example values)
ML = 3.3420E-04;
AL = 0.1;
K1 = 774.8853;
K2 = 1321.0789;

% Read raster
[DN, R] = readgeoraster(thermalBand);
DN = double(DN);

% Step 1: DN to Radiance
radiance = ML .* DN + AL;

% Step 2: Radiance to Brightness Temperature
BT = K2 ./ log((K1 ./ radiance) + 1);

% Step 3: Simple LST estimation (Kelvin to Celsius)
LST = BT - 273.15;

% Save output
outputFile = 'results/LST_Map.tif';
geotiffwrite(outputFile, LST, R);

fprintf('LST processing completed successfully!\n');
